src/DurableTask.Netherite.AzureFunctions/NetheriteProviderFactory.cs

```diff
@@ -131,7 +151,8 @@ NetheriteOrchestrationServiceSettings GetNetheriteOrchestrationServiceSettings(s
                 netheriteSettings.CacheOrchestrationCursors = false; // cannot resume orchestrations in the middle
             }
 
-            netheriteSettings.Validate((name) => this.nameResolver.Resolve(name));
+            // validate the settings and resolve the connections
+            netheriteSettings.Validate(connectionResolver);
```

Comment:
If you are interested, `IOptions` has a validation pattern to it. Probably better left to a different PR (time and priority permitting). But I find it helps with encapsulation, so you can take in `IOptions<T>` and trust that object is fully valid by the time it is added here.

Reply:
I will leave that to a future refactoring in a separate PR.